src/_pytest/python_api.py

```diff
@@ -899,6 +899,13 @@ def raises(
     """
     __tracebackhide__ = True
 
+    if expected_exception == ():
+        raise ValueError(
+            "Passing expected_exception=() is an error, because it's impossible to "
+            "raise an exception which is not an instance of any type. Raising exceptions "
+            "is already understood as failing the test, so you don't need any special "
+            "code to say 'this should never raise an exception'."
+        )
```

Comment:
this message seems a but too elaborate to me, a first idea for a shorter version may be something like (might need another 1-2 iterations)

```suggestion
    if expected_exception == ():
        raise ValueError(
            "The parameter ``expected_exception`` was ``()`` instead of a exception type or a tuple of exception types.\n"
            "Please pass a Exception subclass or a Tuple of them to pytest.raises."
        )
```


it seems surprisingly hard to make this one short and to the point

Reply:
Thanks for the suggestion @RonnyPfannschmidt. should I apply and commit the suggested change, or allow some time for other team members to review and comment on this?
Also, I'm thinking of making the following minor changes to the last line:

```python
"Please pass an Exception subclass or a tuple of them to ``pytest.raises``."
              ^                         ^                ^              ^
```


Please let me know what you think :)
Thanks

Reply:
my suggestion was merely a starting point - maybe @nicoddemus has another insight

your fixed one is certainly better than my first hastily typed suggestion

i propose you do the change, and i approve

if anyone else has another suggestion they can add it

Reply:
The suggestion looks good; while I appreciate the initial take to explain the ins and outs, I think being brief here is a bit better.

Also, I suggest removing the double ticks around the names, as this text is printed to the console, I don't think they make the message easier to read/understand.

Also reading it a few more times, I think we can be even briefer:

```suggestion
    if not expected_exception:
        raise ValueError(
            "Expected an exception type or a tuple a tuple of exceptions types, but got {:r}"
        )
```

The `not expected_exception` will also catch other problems, like passing `None` or `[]`.

Reply:
I'll argue for keeping "Raising exceptions is already understood as failing the test, so you don't need any special  code to say 'this should never raise an exception'."

I originally proposed this as a final sentence because it dissolves a remarkably common beginner misconception, and helps to teach the way that `pytest` works differently from unit testing tools in other languages which often require something like an `assertRaises` method.  @nicoddemus' shorter message is a better opener, but I'd like to add the beginner's advice too.

Reply:
As another data point: I've seen this question (how to make sure it doesn't raise an exception) come up from time to time in my pytest trainings as well. I don't think anyone has tried `()` so far, but I agree it would probably be helpful to add that explanation.

Reply:
No problem, makes sense! Indeed that's a common beginner misconception.